master/buildbot/process/remotecommand.py

```diff
@@ -196,6 +198,34 @@ def interrupt(self, why):
         except Exception as e:
             log.msg("RemoteCommand.interrupt failed", self, e)
 
+    def remote_update_msgpack(self, updates):
+        self.worker.messageReceivedFromWorker()
+        for key, value in updates:
+            try:
+                if self.active and not self.ignore_updates:
+                    if key in ['stdout', 'stderr', 'header']:
+                        whole_line = self.split_line(key, value)
+                        if whole_line is not None:
+                            self.remoteUpdate(key, whole_line, False)
+                    elif key == "log":
+                        logname, data = value
+                        whole_line = self.split_line(logname, data)
+                        value = (logname, whole_line)
+                        if whole_line is not None:
+                            self.remoteUpdate(key, value, False)
+                    else:
+                        self.remoteUpdate(key, value, False)
+            except Exception:
+                # log failure, terminate build, let worker retire the update
+                self._finished(Failure())
```

Comment:
the for will continue. is that intended?

Reply:
Fixed.